### ISSUE TITLE
[RRFC] add bash/zsh completion to npx

### ISSUE THREAD

**Issue by fleytman** [npm/rfcs#557]
## Motivation ("The Why")
It is convenient when you can press tab in the console and get a list of commands. This is how almost all console programs work.

### Example
This could simplify the use of npx. 

#### Current Behaviour
It not working.
#### Desired Behaviour
npx rea -> npx react-native
npx react-native  r -> npx react-native run-
npx react-native run-android --co -> npx react-native run-android --configuration

### References
(https://github.com/npm/cli/issues/4577)

**Comment by ljharb** [comment]
Wouldn't this mean that tab-completion has to contact the registry? `npx` doesn't just run with local and global bins, but with "anything on npmjs.org"

**Comment by lukekarrys** [comment]
I think this could be fairly robust if it used the npx specific cache.

**Comment by quazgar** [comment]
Yes, starting with locally known binaries would already be quite advantageous. Adding a cache for previously successful commands, or whatever, could be added later.
